tests/cases/fourslash/codeFixMapIndexSignatureAll.ts

```diff
@@ -0,0 +1,25 @@
+/// <reference path='fourslash.ts' />
+
+// @target:ES2015
+// @strict:true
+
+// target and lib do not actually work in four slash tests, so we define Map manually
+////declare class Map<K, V> { get(key: K): V; set(key:K, value: V): void; }
+////let map = new Map<string, string>();
+////map['af123'] = "test";
```

Comment:
What does it do for `map['a'] = map['b']`?

Reply:
It works now. It required quite a bit of change since overlapping changes are not supported very well. Although `map['a'] = (map['b'] = map['c']);` now quick fixes to `map.set('a', (map.set('b', map.get('c'))));` which is not correct but I don't think its that common 🤷‍♀️